File: src/selection_sort.cpp
#include<iostream>
#include "../include/selection_sort.hpp"
using namespace std;
template class SelectionSort<int>;
template<class T>
void SelectionSort<T>::selection_sort(int low,int high)
{
        int i,j;
        int min=0;
        for(i=0;i<high;i++)
        {
            min=i;
            for(j=i+1;j<high;j++)
            {
                if(dat[j]<dat[min])
                    min=j;
            }
            if(min!=i)
                swap(dat[min],dat[i]);
        }
}

template<class T>
void SelectionSort<T>::sort_data()
{
    int low,high;
    dat=this->getd();
    l=this->getl();
    low=0;
    high=l-1;
    selection_sort(low,high);
    this->setd(dat,l);
}

File: sorting_python/selectionsort.py
from sort import Sort

class SelectionSort(Sort):

    def selectionsort(self,arr,low,high):
        for i in range(low,high+1):
            min=i
            for j in range(i+1,high+1):
                if arr[j] < arr[min]:
                    min=j
            if min!=i:
                arr[min],arr[i]=arr[i],arr[min]
            print(arr)

    def sort_data(self):
        data=self.getData()
        self.selectionsort(data,0,len(data)-1)
